packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock.tsx

```diff
@@ -1,150 +1,174 @@
+                                )}
+                            </h3>
+                        </EuiText>
+                        <EuiText>Instance ID: {subscriptionInstanceId}</EuiText>
+                    </EuiFlexItem>
+                    <EuiFlexItem grow={false}>
+                        <EuiButtonEmpty
+                            aria-label={
+                                hideDetails
+                                    ? t('showDetails')
+                                    : t('hideDetails')
+                            }
+                            size={'m'}
+                            onClick={() => setHideDetails(!hideDetails)}
+                        >
+                            {hideDetails ? t('showDetails') : t('hideDetails')}
+                        </EuiButtonEmpty>
+                    </EuiFlexItem>
+                </EuiFlexGroup>
+
+                <EuiSpacer size={'m'}></EuiSpacer>
+                {
+                    <table width="100%">
+                        {!hideDetails && (
+                            <tr key={-1}>
+                                <td
+                                    valign={'top'}
+                                    style={{
+                                        width: 250,
+                                        paddingLeft: 0,
+                                        paddingTop: 10,
+                                        paddingBottom: 10,
+                                        borderTop: `solid 1px #ddd`,
+                                        borderBottom: `solid 1px #ddd`,
+                                    }}
+                                >
+                                    <b>Owner subscription ID</b>
+                                </td>
+                                <td
+                                    style={{
+                                        paddingLeft: 0,
+                                        paddingTop: 10,
+                                        paddingBottom: 10,
+                                        borderTop: `solid 1px #ddd`,
+                                        borderBottom: `solid 1px #ddd`,
+                                    }}
+                                >
+                                    {ownerSubscriptionId}
+                                </td>
+                            </tr>
+                        )}
+
+                        {productBlockInstanceValues
+                            .filter(
+                                (productBlockInstanceValue) =>
+                                    !HIDDEN_KEYS.includes(
+                                        productBlockInstanceValue.field,
                                     ),
-                            )}
-                        </table>
-                    }
-                </div>
+                            )
+                            .map((productBlockInstanceValue, index) => (
```

Comment:
Would it make sense to have this as a (function returning a) component instead of deeply nested code?

Reply:
Currently 204 lines, forgot to push the code?

If yes, stop reading my comment here.

Tbh I prefer even smaller components if possible. There are similar blocks of <td> and <tr> in this component that you could extract into components with a name representing where it stands for.